### ISSUE TITLE
Allow favoriting/pinning of courses and course instances on home page

### ISSUE THREAD

**Issue by nicknytko** [PrairieLearn/PrairieLearn#2382]
I realise this is somewhat of a first world problem, but it can be kind of hectic to scroll through courses and instances on the home page if you have a long list of them.  It would be nice if I could "pin" or "favorite" courses to the top so I don't have to scroll.

**Comment by coatless** [comment]
Perhaps there is an underlying issue that should be resolved before pinning.

In particular:

1. There are "courses" and "course instances" shown together for instructors.
2. Past courses are still shown with active courses. 

Perhaps a better flow would be showing the active semester and, then, an "course administrator" mode?

**Comment by nicknytko** [comment]
Or maybe a card at the top showing active courses for this semester (only courses that you are explicitly listed as instructor/ta/student/etc. for, overriding the administrator god-mode), and then the current courses/course instances cards below.

**Comment by nicknytko** [comment]
> Or maybe a card at the top showing active courses for this semester (only courses that you are explicitly listed as instructor/ta/student/etc. for, overriding the administrative god-mode), and then the current courses/course instances cards below.

To add to this, we probably would only want to show an "active courses" list for instructors with multiple courses.  This would probably be redundant for students and instructors of only one course.

@mfsilva22 @mwest1066 thoughts?  I know I can't be the only one that has to scroll through a giant list of courses 🙂

**Comment by coatless** [comment]
@nicknytko I'm worried about streamlining the entire process to save a single click (e.g autoload a single course) when it results in the loss of familiarity as use increases. 

Instructors sometimes help others (vice versa with TAs).

**Comment by trombonekenny** [comment]
There are so many things I'd like to do to change the home page from a list of courses to something more personal and contextual. I have an old POC PR for that page that at least spit out course instances one-course-per-line to help.

Something that's helped (at least how I use PL) has been that you can easily get to course instances via courses, so I almost never search further down into the page. I think it might be a good design to split the "here's the admin list of courses" out from the home page, and let the display logic for that page work the same for admins as instructors (do I have a direct role? am I enrolled?).

We'd need to think about how devMode works - maybe have a different home landing for dev (where dev@illinois.edu isn't part of the course) and production.

**Comment by mwest1066** [comment]
The original issue referred to the problem that admins see all courses on the homepage. Does this still apply for institution admins? I know global admins have this problem. If the problem doesn’t apply to institution admins then I think we should close this issue and let global admins worry about it elsewhere. If institution admins do see all courses on the homepage then I think we should fix that before closing this issue. The fix could be something like what we do in PT, where institution admins have a homepage with only those courses where the admin is explicitly course staff, and admins need to go to the admin pages to find other courses. Ideally we’d have the same setup for global admins but I care less about them 😂

**Comment by nwalters512** [comment]
I'm 99% sure that institution admins just see the institution and have to click into it to see courses, just like PT. I agree that we don't really care about global admins, and in fact I really really like having the homepage being something where I can Cmd-F all courses that exist.

**Comment by mwest1066** [comment]
I agree that institutional admins in PL behave like in PT, where they only see the courses where they are explicitly course staff. Given that, I think we can close this issue.
